Feat/logger function

Volzy opened the pull request:
Added a logger function that can be called with severity.

- [x] Takes a severity argument.
- [x] Takes a message.
- [x] Outputs a colored console.log message based on severity.
- [x] Has tests.
- [ ] Read log level from configuration reader.

Please be critical in this review, @Saturate. My experience with writing tests (and nodejs for that matter) is rather limited.

Volzy committed:
Add logging function

Add log function with severity flag including a test suite to verify if console outputs what's expected.

Volzy committed:
Fix broken test suite

Mocha broke if the test failed, because we have a helper,
that intercepts the console.log messages in order to verify
that output actually happens.

This is handled by moving the logs array outside the console-helper
function

Volzy committed:
Add TODO in order to read ConfigurationReader

Saturate reviewed (changes requested):
Looks really great, but could you fix the ESLint errors?

For now we have to do a `eslint .` in the root, until my PR with eslint is merged.

Volzy committed:
Merge branch 'master' of github.com:Saturate/polymerase into feat/logger-function

Volzy committed:
Add no-console: off to lint and adhear to lint rules

When creating NodeJS apps ESLint don't recommend adding no-console.
Therefore the off setting is added to the linting rule.

Volzy commented:
The lint errors is mainly [no-console](http://eslint.org/docs/rules/no-console). I have disabled it as ESLint specifies:

**When Not To Use It**
If you’re using Node.js, however, console is used to output information to the user and so is not strictly used for debugging purposes. If you are developing for Node.js then you most likely do not want this rule enabled.

@Saturate

Volzy committed:
Merge branch 'master' into feat/logger-function
